include/aws/common/string.h

```diff
@@ -289,23 +206,17 @@ AWS_EXTERN_C_END
  * accordingly. If there is insufficient space in the buf, returns
  * false, leaving the buf unchanged.
  */
-AWS_STATIC_IMPL bool aws_byte_buf_write_from_whole_string(
+AWS_COMMON_API
+bool aws_byte_buf_write_from_whole_string(
     struct aws_byte_buf *AWS_RESTRICT buf,
-    const struct aws_string *AWS_RESTRICT src) {
-    AWS_PRECONDITION(!buf || aws_byte_buf_is_valid(buf));
-    AWS_PRECONDITION(!src || aws_string_is_valid(src));
-    if (buf == NULL || src == NULL) {
-        return false;
-    }
-    return aws_byte_buf_write(buf, aws_string_bytes(src), src->len);
-}
+    const struct aws_string *AWS_RESTRICT src);
 
 /**
  * Creates an aws_byte_cursor from an existing string.
  */
-AWS_STATIC_IMPL struct aws_byte_cursor aws_byte_cursor_from_string(const struct aws_string *src) {
-    AWS_PRECONDITION(aws_string_is_valid(src));
-    return aws_byte_cursor_from_array(aws_string_bytes(src), src->len);
-}
+AWS_COMMON_API
+struct aws_byte_cursor aws_byte_cursor_from_string(const struct aws_string *src);
```

Comment:
I could go either way on this one, it's so trivial. I can't make up my mind.

Reply:
I didn't move this one to the .inl, because initializers are unlikely to be on the critical path.